internal/provider/resource_tfe_team_notification_configuration.go

```diff
@@ -0,0 +1,458 @@
+			"id": schema.StringAttribute{
+				Description: "ID of the team notification configuration.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"name": schema.StringAttribute{
+				Description: "Name of the team notification configuration.",
+				Required:    true,
+			},
+
+			"destination_type": schema.StringAttribute{
+				Description: "The type of notification configuration payload to send.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf(
+						string(tfe.NotificationDestinationTypeEmail),
+						string(tfe.NotificationDestinationTypeGeneric),
+						string(tfe.NotificationDestinationTypeSlack),
+						string(tfe.NotificationDestinationTypeMicrosoftTeams),
+					),
+				},
+			},
+
+			"email_addresses": schema.SetAttribute{
+				Description: "A list of email addresses. This value must not be provided if `destination_type` is `generic`, `microsoft-teams`, or `slack`.",
+				Optional:    true,
+				Computed:    true,
+				ElementType: types.StringType,
+				Validators: []validator.Set{
+					validators.AttributeValueConflictSetValidator(
+						"destination_type",
+						[]string{"generic", "microsoft-teams", "slack"},
+					),
+					setvalidator.ConflictsWith(
+						path.MatchRelative().AtParent().AtName("token"),
+						path.MatchRelative().AtParent().AtName("url"),
+					),
+				},
+			},
+
+			"email_user_ids": schema.SetAttribute{
+				Description: "A list of user IDs. This value must not be provided if `destination_type` is `generic`, `microsoft-teams`, or `slack`.",
+				Optional:    true,
+				Computed:    true,
+				ElementType: types.StringType,
+				Validators: []validator.Set{
+					validators.AttributeValueConflictSetValidator(
+						"destination_type",
+						[]string{"generic", "microsoft-teams", "slack"},
+					),
+					setvalidator.ConflictsWith(
+						path.MatchRelative().AtParent().AtName("token"),
+						path.MatchRelative().AtParent().AtName("url"),
+					),
+				},
+			},
+
+			"enabled": schema.BoolAttribute{
+				Description: "Whether the team notification configuration should be enabled or not. Disabled configurations will not send any notifications. Defaults to `false`.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+
+			"token": schema.StringAttribute{
+				Description: "A write-only secure token for the notification configuration, which can be used by the receiving server to verify request authenticity when configured for notification configurations with a destination type of `generic`. Defaults to `null`. This value _must not_ be provided if `destination_type` is `email`, `microsoft-teams`, or `slack`.",
+				Optional:    true,
+				Sensitive:   true,
+				Validators: []validator.String{
```

Comment:
We should add the setvalidator to conflict with `email_addresses` and`email_user_id` here?

Reply:
my preference is to only have conflicts in one place, otherwise the UX is kind of obnoxious. the validations were in a somewhat inconsistent state, so I consolidated them more

Reply:
Okay! sounds good